ARE/components/actuator.irtrans/src/main/java/eu/asterics/component/actuator/irtrans/IrTransInstance.java

```diff
@@ -294,43 +318,43 @@ public void start() {
      */
     @Override
     public void stop() {
-        closeSocket();
+        closeTcpSocket(tcpSocket);
+        closeTcpSocket(tcpSocketRead);
+        readerThread.interrupt();
         AstericsErrorHandling.instance.reportInfo(this, "IRTransmitter Instance stopped");
         super.stop();
     }
 
-    private void closeSocket() {
+    private void closeTcpSocket(Socket socket) {
         if (socket != null) {
             try {
                 socket.close();
             } catch (Exception e) {
-                logger.log(Level.WARNING, "error closing socket", e);
+                logger.log(Level.WARNING, "error closing tcpSocket", e);
             }
         }
-        socket = null;
     }
 
-    private String readFromSocket(int timeoutMillis) {
+    private String readFromSocket(Socket socket, int timeoutMillis) {
         String result = ERROR_RESULT;
         try {
             socket.setSoTimeout(timeoutMillis);
             BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
             result = bufferedReader.readLine();
         } catch (SocketTimeoutException e) {
-            logger.log(Level.WARNING,
-                    MessageFormat.format("reading from socket timed out (timeout: {0}ms)", timeoutMillis), e);
+            logger.log(Level.WARNING, MessageFormat.format("reading from tcpSocket timed out (timeout: {0}ms)", timeoutMillis), e);
         } catch (IOException e1) {
-            logger.log(Level.WARNING, "exception reading from socket", e1);
-            closeSocket();
+            logger.log(Level.WARNING, "exception reading from tcpSocket", e1);
+            closeTcpSocket(socket);
         }
         return result;
     }
 
-    private void sendToSocket(String cmdString) throws IOException {
+    private void sendToSocket(Socket socket, String cmdString) throws IOException {
         // remove spaces at beginning and end and replace double spaces in
         // string with single space
         byte[] sendBytes = ("A" + cmdString.trim().replaceAll("\\s+", " ") + "\n").getBytes();
-        socket.getInputStream().skip(socket.getInputStream().available());
+        socket.getInputStream().skip(tcpSocket.getInputStream().available());
```

Comment:
you directly use the member variable tcpSocket, but shouldnt it be the parameter socket?